frontend/app/components/progress.tsx

```diff
@@ -26,13 +27,19 @@ const indicatorBaseClassName = 'h-full w-full flex-1 transition-all';
 export interface ProgressProps extends React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> {
   size?: keyof typeof sizes;
   variant?: keyof typeof variants;
+  label: string;
+  locale: string;
+  value: number;
 }
 
-const Progress = React.forwardRef<React.ElementRef<typeof ProgressPrimitive.Root>, ProgressProps>(({ className, size = 'base', variant = 'default', value, ...props }, ref) => {
+const Progress = React.forwardRef<React.ElementRef<typeof ProgressPrimitive.Root>, ProgressProps>(({ className, size = 'base', variant = 'default', value, locale, label, ...props }, ref) => {
   return (
-    <ProgressPrimitive.Root ref={ref} className={cn(rootBaseClassName, sizes[size], className)} data-testid="progress-root" value={value} {...props}>
-      <ProgressPrimitive.Indicator className={cn(indicatorBaseClassName, variants[variant])} style={{ transform: `translateX(-${100 - (value ?? 0)}%)` }} data-testid="progress-indicator" />
-    </ProgressPrimitive.Root>
+    <>
+      {label && <p id="progress-label">{`${label} ${formatPercent(value, locale)}`}</p>}
```

Comment:
Add the following class to the `progress-label`.

```suggestion
      {label && <p id="progress-label" className="mb-2">{`${label} ${formatPercent(value, locale)}`}</p>}
```